uaclient/system.py

```diff
@@ -282,15 +283,19 @@ def get_machine_id(cfg) -> str:
         if cfg_machine_id:
             return cfg_machine_id
 
-    fallback_machine_id_file = cfg.data_path("machine-id")
+    fallback_machine_id = machine_id_file.read()
 
-    for path in [ETC_MACHINE_ID, DBUS_MACHINE_ID, fallback_machine_id_file]:
+    for path in [ETC_MACHINE_ID, DBUS_MACHINE_ID]:
         if os.path.exists(path):
             content = load_file(path).rstrip("\n")
             if content:
                 return content
+
+        if fallback_machine_id:
+            return fallback_machine_id
```

Comment:
I think this needs to be outside the loop to keep consistent behavior
```suggestion
    if fallback_machine_id:
        return fallback_machine_id
```